includes/wizards/class-subscriptions-wizard.php

```diff
@@ -0,0 +1,392 @@
+<?php
+/**
+ * Newspack subscriptions setup and management.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+use \WC_Subscriptions_Product, \WC_Product_Simple, \WC_Product_Subscription, \WC_Name_Your_Price_Helpers;
+
+defined( 'ABSPATH' ) || exit;
+
+require_once NEWSPACK_ABSPATH . '/includes/wizards/class-wizard.php';
+
+/**
+ * Easy interface for managing subscriptions.
+ */
+class Subscriptions_Wizard extends Wizard {
+
+	/**
+	 * The name of this wizard.
+	 *
+	 * @var string
+	 */
+	protected $name = 'Subscriptions';
+
+	/**
+	 * The slug of this wizard.
+	 *
+	 * @var string
+	 */
+	protected $slug = 'newspack-subscriptions-wizard';
+
+	/**
+	 * The capability required to access this wizard.
+	 *
+	 * @var string
+	 */
+	protected $capability = 'edit_products';
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		parent::__construct();
+
+		add_action( 'rest_api_init', [ $this, 'register_api_endpoints' ] );
+		add_filter( 'woocommerce_product_data_store_cpt_get_products_query', [ $this, 'handle_only_get_newspack_subscriptions_query' ], 10, 2 );
```

Comment:
[Question] What's the rationale for hooking in like this as opposed to querying for all products and then filtering the results? This approach is clearly a little more efficient I'm just wondering if it introduces any unnecessary risk by altering another plugin's activity?

Reply:
This is [the official supported way of extending `wc_get_products`](https://github.com/woocommerce/woocommerce/wiki/wc_get_products-and-WC_Product_Query#adding-custom-parameter-support), so it should be low-risk. All minor WooCommerce releases (3.x.x) are fully backwards-compatible, so this method is supported until at least WC 4.0.